code/modules/mob/living/carbon/human/human_defense.dm

```diff
@@ -162,10 +162,21 @@ emp_act
 		return TRUE
 
 /mob/living/carbon/human/emp_act(severity)
-	for(var/obj/O in src)
-		if(!O)	continue
+	if(isSynthetic())

```

Comment:
Overriding the emp_act on the IPC level is a much better way do to this than using an istype check, IMO

Reply:
IPCs aren't a subtype they're a species

Reply:
Issue with that is that it'll still do the rest below it. And you don't want that to happen this time.
Issue I see here with the PR code is that it does not go through the object list anymore. Does it still EMP the objects the IPC holds? Like the backpack stuff etc?

Reply:
Right, I forgot how exactly species stuff is set up. Still, the current approach seems not very OOP.

Reply:
Ack, you're right, I screwed up my logic, and forgot to include EMPing the IPC's contents

Reply:
This is exactly the kind of snowflake code that makes a lot of coders despise IPCs to begin with; this is a no go how this is being handled.

Reply:
So uhh.. how should it be handled?

Reply:
Maybe have the species listen to the emp_act signal and act accordingly (signal prob needs to be made)